### DISCUSSION TITLE
How does fyne obtain the device model or serial number of an Android device?

### DISCUSSION THREAD

**Discussion by cplasfwst** [fyne-io/fyne#4687] [Q&A]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

How does fyne obtain the device model or serial number of an Android device?

### Is it possible to construct a solution with the existing API?

Not clear yet

### Describe the solution you'd like to see.

How can I obtain device-related information after packaging it into Android? The information I need is unique.

**Comment by Jacalz** [comment]
Please use discussions for questions. The bug tracker is for feature requests and bug reports.
Please also stick to English as the language as that is the language the all of our team understand. I will transfer this to a discussion for you and translate the text so everyone can understand. Thanks for understanding.

**Comment by andydotxyz** [comment]
The basic answer is to use `driver.RunNative` to execute some CGo code that can ask for this information from the Android NDK.
I have a TODO item to write some of this up, but hopefully that helps in the meantime.

**Reply by cplasfwst** [reply]
Thank you author for your patient answer. Is there any case of driver.RunNative? I'm not very good at it, so I'm asking for your guidance.

**Reply by andydotxyz** [reply]
I have written up a summary of a customer conversation in case it helps https://fynelabs.com/2024/03/01/running-native-android-code-in-a-fyne-app/

**Reply by cplasfwst** [reply]
Thank you author for your patient support! I also have a question, how or what app version do I make for the current program? I want to check and then prompt the user to update, for example the program version I have defined in fyneApp.toml, how do I get it?

**Reply by andydotxyz** [reply]
I think you mean `App.Metadata().Version` https://docs.fyne.io/api/v2.4/appmetadata.html.

Please don't use closed tickets for discussions though - join one of our servers, or use the GitHub discussion feature.

**Reply by andydotxyz** [reply]
This might be relevant too https://fynelabs.com/2022/11/17/easy-selfupdate-deployment-for-go-application/
